skrf/network.py

```diff
@@ -6564,19 +6774,40 @@ def impedance_mismatch(z1: NumberLike, z2: NumberLike) -> npy.ndarray:
             complex impedance of port 1
     z2 : number or array-like
             complex impedance of port 2
+    s_def : str, optional. Default is 'traveling'.
+        Scattering parameter definition:
+        'pseudo' for pseudo-waves definition.
+        'traveling' corresponds to the initial implementation.
+        'power' is not supported.
+        NB: results are the same for real-valued characteristic impedances.
 
     Returns
     -------
     s' : 2-port s-matrix for the impedance mis-match
+
+    References
+    ----------
+    .. [#] R. B. Marks et D. F. Williams, A general waveguide circuit theory,
+            J. RES. NATL. INST. STAN., vol. 97, no. 5, p. 533, sept. 1992.
     """
     from .tlineFunctions import zl_2_Gamma0
     gamma = zl_2_Gamma0(z1, z2)
     result = npy.zeros(shape=(len(gamma), 2, 2), dtype='complex')
 
-    result[:, 0, 0] = gamma
-    result[:, 1, 1] = -gamma
-    result[:, 1, 0] = (1 + gamma) * npy.sqrt(1.0 * z1 / z2)
-    result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
+    if s_def == 'traveling':
+        result[:, 0, 0] = gamma
+        result[:, 1, 1] = -gamma
+        result[:, 1, 0] = (1 + gamma) * npy.sqrt(1.0 * z1 / z2)
+        result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
+    elif s_def == 'pseudo':
+        n = npy.abs(z2/z1) * npy.sqrt(z1.real / z2.real)
+        result[:, 0, 0] = gamma
+        result[:, 1, 1] = -gamma
+        result[:, 1, 0] = 2 * z2 / (n * (z1 + z2))
+        result[:, 0, 1] = 2 * z1 * n / (z1 + z2)
+    else:
+        raise ValueError('Unsupported s_def: {}'.format(s_def))
+
```

Comment:
Actually, it is possible to derive the S-param of a mismatch using the following definitions:
![image](https://user-images.githubusercontent.com/4642848/167574119-24b92b0d-a11b-41d2-a1ae-91f0898aa5a0.png)

Reply:
I don't understand your answer. The equations above correpond to the S-parameters for a mismatch for the power wave definition, which is currently not supported in the `impedance_mismatch`. This is related with the if sections in `connect`: 

https://github.com/scikit-rf/scikit-rf/blob/9f5a57dc2ccb99367b8d317e07772ac13ba46d0a/skrf/network.py#L4015-L4031

Reply:
I didn't try it, but I don't think it solves the problem that power waves with complex ports can't be composed because wave leaving the port is not identical to the wave coming into port that is connected to it, even if the port impedances match. The connection algorithm depends on this continuity.

Reply:
Sure, I didn't pretend it solves the problem of cascading (it can't be in fact), but this was just to provide a solution to `impedance_mismatch` for power waves, even it is not used by the `connect` algorithm (just in case).